### ISSUE TITLE
[regression/8.0.80] [Android] ObjectDisposedException with AndroidX.AppCompat.Widget.AppCompatImageView

### ISSUE THREAD

**Issue by janusw** [dotnet/maui#24757]
### Description

Starting with MAUI 8.0.80 I see ObjectDisposedExceptions on Android when switching shell tabs, where one page includes an `<Image>` inside a `<Frame>`. This worked well up to MAUI 8.0.72.

### Steps to Reproduce

1. Build and run the `ShellTabBarSample` project on this branch of my maui-samples fork: https://github.com/janusw/maui-samples/tree/Frame_Image_bug_repro/8.0/Navigation/ShellTabBarSample
2. Click the button on the first tab.
3. Switch to the second tab.
4. Switch back to the first tab.
5. This exception is thrown:

```
System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'AndroidX.AppCompat.Widget.AppCompatImageView'.
  at Java.Interop.JniPeerMembers.AssertSelf(IJavaPeerable self) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/src/Java.Interop/Java.Interop/JniPeerMembers.cs:153
  at Java.Interop.JniPeerMembers.JniInstanceMethods.InvokeVirtualVoidMethod(String encodedMember, IJavaPeerable self, JniArgumentValue* parameters) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/src/Java.Interop/Java.Interop/JniPeerMembers.JniInstanceMethods_Invoke.cs:57
  at Android.Views.View.RemoveOnAttachStateChangeListener(IOnAttachStateChangeListener listener) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/obj/Release/net8.0/android-34/mcw/Android.Views.View.cs:21313
  at Android.Views.View.<>c__DisplayClass2963_0.<remove_ViewAttachedToWindow>b__0(IOnAttachStateChangeListener __v) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/obj/Release/net8.0/android-34/mcw/Android.Views.View.cs:24196
  at Java.Interop.EventHelper.RemoveEventHandler[IOnAttachStateChangeListener,IOnAttachStateChangeListenerImplementor](WeakReference& implementor, Func`2 empty, Action`1 unsetListener, Action`1 remove) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Java.Interop/EventHelper.cs:35
  at Android.Views.View.remove_ViewAttachedToWindow(EventHandler`1 value) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/obj/Release/net8.0/android-34/mcw/Android.Views.View.cs:24193
  at Microsoft.Maui.Handlers.ImageHandler.DisconnectHandler(ImageView platformView) in /_/src/Core/src/Handlers/Image/ImageHandler.Android.cs:33
  at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.IImage, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Android.Widget.ImageView, Mono.Android, Version=0.0.0.0, Culture=neutral, PublicKeyToken=84e04ff9cfb79065]].OnDisconnectHandler(View platformView) in /_/src/Core/src/Handlers/View/ViewHandlerOfT.cs:79
  at Microsoft.Maui.Handlers.ViewHandler.OnDisconnectHandler(Object platformView) in /_/src/Core/src/Handlers/View/ViewHandler.cs:214
  at Microsoft.Maui.Handlers.ElementHandler.DisconnectHandler(Object platformView) in /_/src/Core/src/Handlers/Element/ElementHandler.cs:112
  at Microsoft.Maui.Handlers.ElementHandler.Microsoft.Maui.IElementHandler.DisconnectHandler() in /_/src/Core/src/Handlers/Element/ElementHandler.cs:130
  at Microsoft.Maui.Controls.Element.SetHandler(IElementHandler newHandler) in /_/src/Controls/src/Core/Element/Element.cs:984
  at Microsoft.Maui.Controls.Element.set_Handler(IElementHandler value) in /_/src/Controls/src/Core/Element/Element.cs:929
  at Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IElement.set_Handler(IElementHandler value) in /_/src/Controls/src/Core/VisualElement/VisualElement.cs:2074
  at Microsoft.Maui.Platform.ElementExtensions.ToHandler(IElement view, IMauiContext context) in /_/src/Core/src/Platform/ElementExtensions.cs:96
  at Microsoft.Maui.Platform.ElementExtensions.ToPlatform(IElement view, IMauiContext context) in /_/src/Core/src/Platform/ElementExtensions.cs:127
  at Microsoft.Maui.Controls.Handlers.Compatibility.FrameRenderer.UpdateContent() in /_/src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs:354
  at Microsoft.Maui.Controls.Handlers.Compatibility.FrameRenderer.<>c.<.cctor>b__73_6(FrameRenderer h, Frame _) in /_/src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs:28
  at Microsoft.Maui.PropertyMapper`2.<>c__DisplayClass5_0[[Microsoft.Maui.Controls.Frame, Microsoft.Maui.Controls, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Controls.Handlers.Compatibility.FrameRenderer, Microsoft.Maui.Controls, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].<Add>b__0(IElementHandler h, IElement v) in /_/src/Core/src/PropertyMapper.cs:172
  at Microsoft.Maui.PropertyMapper.UpdatePropertyCore(String key, IElementHandler viewHandler, IElement virtualView) in /_/src/Core/src/PropertyMapper.cs:47
  at Microsoft.Maui.PropertyMapper.UpdateProperties(IElementHandler viewHandler, IElement virtualView) in /_/src/Core/src/PropertyMapper.cs:82
  at Microsoft.Maui.Controls.Handlers.Compatibility.VisualElementRenderer`1[[Microsoft.Maui.Controls.Frame, Microsoft.Maui.Controls, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IElement view, IPlatformViewHandler nativeViewHandler, Action`1 onElementChanged, Frame& currentVirtualView, IPropertyMapper& _mapper, IPropertyMapper _defaultMapper, Boolean autoPackage) in /_/src/Controls/src/Core/Compatibility/Handlers/VisualElementRenderer.cs:290
  at Microsoft.Maui.Controls.Handlers.Compatibility.ViewHandlerDelegator`1[[Microsoft.Maui.Controls.Frame, Microsoft.Maui.Controls, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IElement view, Action`1 onElementChanged, Boolean autoPackage) in /_/src/Controls/src/Core/Compatibility/Handlers/ViewHandlerDelegator.cs:96
  at Microsoft.Maui.Controls.Handlers.Compatibility.FrameRenderer.Microsoft.Maui.IElementHandler.SetVirtualView(IElement view) in /_/src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs:405
  at Microsoft.Maui.Controls.Element.SetHandler(IElementHandler newHandler) in /_/src/Controls/src/Core/Element/Element.cs:987
  at Microsoft.Maui.Controls.Element.set_Handler(IElementHandler value) in /_/src/Controls/src/Core/Element/Element.cs:929
  at Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IElement.set_Handler(IElementHandler value) in /_/src/Controls/src/Core/VisualElement/VisualElement.cs:2074
  at Microsoft.Maui.Platform.ElementExtensions.ToHandler(IElement view, IMauiContext context) in /_/src/Core/src/Platform/ElementExtensions.cs:96
  at Microsoft.Maui.Platform.ElementExtensions.ToPlatform(IElement view, IMauiContext context) in /_/src/Core/src/Platform/ElementExtensions.cs:127
  at Microsoft.Maui.Handlers.ContentViewHandler.UpdateContent(IContentViewHandler handler) in /_/src/Core/src/Handlers/ContentView/ContentViewHandler.Android.cs:42
  at Microsoft.Maui.Handlers.ContentViewHandler.MapContent(IContentViewHandler handler, IContentView page) in /_/src/Core/src/Handlers/ContentView/ContentViewHandler.Android.cs:47
  at Microsoft.Maui.PropertyMapper`2.<>c__DisplayClass5_0[[Microsoft.Maui.IContentView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Handlers.IContentViewHandler, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].<Add>b__0(IElementHandler h, IElement v) in /_/src/Core/src/PropertyMapper.cs:172
  at Microsoft.Maui.PropertyMapper.UpdatePropertyCore(String key, IElementHandler viewHandler, IElement virtualView) in /_/src/Core/src/PropertyMapper.cs:47
  at Microsoft.Maui.PropertyMapper.UpdateProperties(IElementHandler viewHandler, IElement virtualView) in /_/src/Core/src/PropertyMapper.cs:82
  at Microsoft.Maui.Handlers.ElementHandler.SetVirtualView(IElement view) in /_/src/Core/src/Handlers/Element/ElementHandler.cs:79
  at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.IContentView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.ContentViewGroup, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IView view) in /_/src/Core/src/Handlers/View/ViewHandlerOfT.cs:53
  at Microsoft.Maui.Handlers.ContentViewHandler.SetVirtualView(IView view) in /_/src/Core/src/Handlers/ContentView/ContentViewHandler.Android.cs:26
  at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.IContentView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.ContentViewGroup, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IElement view) in /_/src/Core/src/Handlers/View/ViewHandlerOfT.cs:56
  at Microsoft.Maui.Controls.Element.SetHandler(IElementHandler newHandler) in /_/src/Controls/src/Core/Element/Element.cs:987
  at Microsoft.Maui.Controls.Element.set_Handler(IElementHandler value) in /_/src/Controls/src/Core/Element/Element.cs:929
  at Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IElement.set_Handler(IElementHandler value) in /_/src/Controls/src/Core/VisualElement/VisualElement.cs:2074
  at Microsoft.Maui.Platform.ElementExtensions.ToHandler(IElement view, IMauiContext context) in /_/src/Core/src/Platform/ElementExtensions.cs:96
  at Microsoft.Maui.Platform.ViewExtensions.ToHandler(IView view, IMauiContext context) in /_/src/Core/src/Platform/ViewExtensions.cs:44
  at Microsoft.Maui.Controls.Platform.Compatibility.ShellContentFragment.OnCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) in /_/src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellContentFragment.cs:145
  at AndroidX.Fragment.App.Fragment.n_OnCreateView_Landroid_view_LayoutInflater_Landroid_view_ViewGroup_Landroid_os_Bundle_(IntPtr jnienv, IntPtr native__this, IntPtr native_inflater, IntPtr native_container, IntPtr native_savedInstanceState) in C:\a\_work\1\s\generated\androidx.fragment.fragment\obj\Release\net6.0-android\generated\src\AndroidX.Fragment.App.Fragment.cs:2031
  at Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PPLLL_L(_JniMarshal_PPLLL_L callback, IntPtr jnienv, IntPtr klazz, IntPtr p0, IntPtr p1, IntPtr p2) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Android.Runtime/JNINativeWrapper.g.cs:363
```

### Link to public reproduction project repository

https://github.com/janusw/maui-samples/tree/Frame_Image_bug_repro/8.0/Navigation/ShellTabBarSample

### Version with bug

8.0.80 SR8

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

8.0.71 SR7.1

### Affected platforms

Android

### Affected platform versions

seen on Android 14

### Did you find any workaround?

revert to MAUI 8.0.72

**Comment by QianaJiao** [comment]
When verifying https://github.com/dotnet/maui/issues/24555, I saw the same error, they may be caused by the same issue.
![image](https://github.com/user-attachments/assets/983a5f0b-5f92-4183-9b05-7762e8d424ef)
[DemoApp.zip](https://github.com/user-attachments/files/16978111/DemoApp.zip)

Repro on 8.0.80, not repro on 8.0.72

**Comment by mattleibow** [comment]
@PureWeen i see disconnect handler in there, so maybe we are missing an is alive check. We probably never hit this because no one ever called disconnect before.

**Comment by janusw** [comment]
> When verifying #24555, I saw the same error, they may be caused by the same issue.

Yes, I think it's the same underlying issue. That one also has an `Image` inside a `Frame` and it also occurs when navigating (on Android).


> Repro on 8.0.80, not repro on 8.0.72

:+1: 

(Btw I also see it with 8.0.82.)

This really is a showstopper for us that keeps us from updating to 8.0.8x.

**Comment by mattleibow** [comment]
Wait a sec... @PureWeen added an opt-out feature for the auto disconnect to get the old behavior. Let me find the PR and get some code

**Comment by mattleibow** [comment]
https://github.com/dotnet/maui/pull/23738

Try using this to prevent the force disconnect for the time being.

**Comment by mattleibow** [comment]
Ah, that is net9.0 only. Maybe we need to add this in net 8.0 somehow @PureWeen?

**Comment by janusw** [comment]
Just saw that 8.0.90 has been released, tried it, and it actually seems to fix this issue (at least for my reproducer above)! 🥳

**Comment by janusw** [comment]
> Just saw that 8.0.90 has been released, tried it, and it actually seems to fix this issue (at least for my reproducer above)! 🥳

I also verified that it fixes the problem not only in the reduced sample, but also in my real-world use case. :+1:

**Comment by mattleibow** [comment]
That is very good to hear! Can I close this issue?

**Comment by janusw** [comment]
> That is very good to hear! Can I close this issue?

Yes, the issue is fixed via MAUI version 8.0.90 for me. Closing.
